File: http.js
const http = require('http');
const server = http.createServer();

const sleep = async ms => {
    return new Promise(resolve => {
        setTimeout(() => resolve(), ms);
    });
};

server.on('request', async function(req, res) {
    await sleep(3000);
    res.writeHead(200, {'Content-Type' : 'text/plain'});
    res.write('Hello');
    res.end();
});

server.listen(3000);

File: http.rb
require 'socket'

server = TCPServer.new 3000
loop do
  client = server.accept
  headers = []
  loop do
    header = client.gets.chomp
    break if header.empty?

    headers << header
  end
  p headers

  sleep 3

  client.puts 'HTTP/1.0 200 OK'
  client.puts 'Content-Type: text/plain'
  client.puts
  client.puts 'Hello'
  client.close
end

